packages/beacon-node/src/chain/shufflingCache.ts

```diff
@@ -19,10 +19,14 @@ import {Metrics} from "../metrics/metrics.js";
 const MAX_EPOCHS = 4;
 
 /**
+ * OLD NOTE:
  * With default chain option of maxSkipSlots = 32, there should be no shuffling promise. If that happens a lot, it could blow up Lodestar,
  * with MAX_EPOCHS = 4, only allow 2 promise at a time. Note that regen already bounds number of concurrent requests at 1 already.
+ * 
+ * UPDATE:
+ * This blew up on non-finality devnet.  Up number to 4 to allow for additional forks and old blocks coming across network.
  */
-const MAX_PROMISES = 2;
+const MAX_PROMISES = 4;
```

Comment:
I don't think we want to increase it, it'll hide the root cause of the issue

currently, we always request to build a shuffling async which is not great at syncing time. We need to change our behavior like:
- at syncing time, do not request async shuffling, always do it in `afterProcessEpoch()` so that it works as before
- at synced time, go with the current behavior, ie request async build at `preparedNextSlot()`